Addition of new design considerations

ianbjacobs opened the pull request:
Taken from
https://github.com/w3c/webpayments/wiki/PaymentApp_Notes#design-consider
ations

ianbjacobs committed:
Addition of new design considerations

Taken from
https://github.com/w3c/webpayments/wiki/PaymentApp_Notes#design-consider
ations

halindrome commented on index.html:
```diff
@@ -258,7 +258,7 @@
 <section id="model">
   <h2>Payment App Model and Design Considerations</h2>
-  <p>This section (which may be temporary) intends to help build shared understanding of the capabilities and limitations of the specified model.</p>
+  <p>This section (which maypaym be temporary) intends to help build shared understanding of the capabilities and limitations of the specified model.</p>
```
maypaym ?

halindrome commented on index.html:
```diff
@@ -316,12 +363,18 @@
           asynchronous mechanism for a payment app to return a payment response to the user agent. This specification
-          does not address scenarios where the user agent does not receive the response (e.g., due to a crash our
-          network outage).</li>
+          does not (yet) address scenarios where the user agent does not receive the response (e.g., due to a crash our
```
crash _or_

halindrome commented on index.html:
```diff
@@ -316,12 +363,18 @@
-          does not address scenarios where the user agent does not receive the response (e.g., due to a crash our
-          network outage).</li>
+          does not (yet) address scenarios where the user agent does not receive the response (e.g., due to a crash our
+          network outage). Question: Should we recommend asynchronous communication between payment app and servers it talks to?</li>
```
Instead of embedding this question, I suggest you raise an issue and then put an issue marker in the spec.

adrianhopebailie commented on index.html:
```diff
@@ -269,19 +269,66 @@
+    <h3>Decoupling and Trust</h3>
+    <ul>
+<li>    A goal of this system is to decouple the payment methods used to pay from the software (payment apps) that implement those payment methods. By decoupling, merchants (and their payment service providers) can lower the cost of creating a checkout experience, users can have more choice in the software they use to pay, and payment app providers can innovate without imposing an integration bur ...
+<li>    Users may choose to use "open" or "proprietary" payment methods, so the payment app ecosystem must support both. Users must be able to register payment apps of their choosing. We expect the user to have greater choice of third party payment apps for open payment methods than for proprietary payment methods. Examples of open payment methods include card payment and SEPA credit transfer.</l ...
```
👍

adrianhopebailie commented on index.html:
```diff
@@ -269,19 +269,66 @@
   <section>
+    <h3>Decoupling and Trust</h3>
+    <ul>
+<li>    A goal of this system is to decouple the payment methods used to pay from the software (payment apps) that implement those payment methods. By decoupling, merchants (and their payment service providers) can lower the cost of creating a checkout experience, users can have more choice in the software they use to pay, and payment app providers can innovate without imposing an integration bur ...
```
👍

adrianhopebailie commented on index.html:
```diff
@@ -269,19 +269,66 @@
+    <ul>
+<li>    A goal of this system is to decouple the payment methods used to pay from the software (payment apps) that implement those payment methods. By decoupling, merchants (and their payment service providers) can lower the cost of creating a checkout experience, users can have more choice in the software they use to pay, and payment app providers can innovate without imposing an integration bur ...
+<li>    Users may choose to use "open" or "proprietary" payment methods, so the payment app ecosystem must support both. Users must be able to register payment apps of their choosing. We expect the user to have greater choice of third party payment apps for open payment methods than for proprietary payment methods. Examples of open payment methods include card payment and SEPA credit transfer.</l ...
+<li>    For privacy, the design should limit information about the user's environment available to merchant without user consent. That includes which payment apps the user has registered. For open payment methods, the merchant should not receive information about which payment app the user selected to pay unless the user consents to share that information.</li>
```
"For open payment methods, the merchant should not receive information about which payment app the user selected to pay unless the user consents to share that information." - this may be difficult because the merchant needs to track the progress of, especially a push payment, with some entity (i.e. knowing the payment method may not be enough)

adrianhopebailie commented on index.html:
```diff
@@ -269,19 +269,66 @@
+<li>    For privacy, the design should limit information about the user's environment available to merchant without user consent. That includes which payment apps the user has registered. For open payment methods, the merchant should not receive information about which payment app the user selected to pay unless the user consents to share that information.</li>
+<li>    Although decoupling relieves merchants of implementing some aspects of the checkout experience, one consequence is that they give up some degree of control. This was already the case for proprietary payment methods, but for open payment methods such as cards, merchants (or their PSPs) will be entrusting some portion of data collection to third party payment apps, without knowing which app ...
+<li>    We recognize, therefore, a tension between user preferences (e.g., registered payment apps, ordering of display of payment apps, etc.) and merchant preferences (e.g., control of the user experience previously implemented in a Web page, ordering of payment apps, presence of merchant's own payment app, etc.).</li>
+<li>    The design should address this tension by enabling the merchant to express preferences for both payment methods and payment apps. The design should not constrain how browsers make us of these preferences, only provide guidance to browser vendors about taking into account both merchant and user preferences.</li>
```
"make _use_"

adrianhopebailie commented on index.html:
```diff
@@ -269,19 +269,66 @@
+  </ul>
+</li>
+<li> The browser uses this information in conjunction with user preferences to:
+     <li>   Filter matching payment apps (in the case of accepted payment apps)</li>
```
matching payment apps are also filtered based on accepted payment methods

adrianhopebailie commented on index.html:
```diff
@@ -269,19 +269,66 @@
+<li>        When the browser recommends a payment app and the user selects it, registration can happen in that moment without additional user action. (There is an expectation that the display of browser-recommended apps will differ from the display of merchant-recommended apps).</li>
+<li>        When the user installs native payment app, registration can happen as part of the app installation process.</li>
+<li>        Users visiting a Web site (e.g., merchant or bank) may wish to explicitly register payment apps.</li>
+	</ul>
```
``` html
<li>        Users visiting a Web site (e.g., merchant or bank) may wish 
to explicitly register payment apps unrelated to a specific checkout 
or payment.</li>
```

adrianhopebailie commented on index.html:
```diff
@@ -269,19 +269,66 @@
           The user agent stores this information for subsequent actions (e.g., when matching merchant-accepted payment methods). In this
           proposal, there are no requirements for a payment app to be able to respond to user agent queries for updated
-          registration information. We expect payment apps to update the information that a user agent has stored about them
+        registration information. In this proposal, payment apps to update the information that a user agent has stored about them
```
"In this proposal, payment apps update the information "

ianbjacobs committed:
Merge remote-tracking branch 'w3c/gh-pages' into gh-pages

ianbjacobs committed:
Small changes based on Shane and AHB feedback

* Fixed link to issues list
* Added two links to issue 224 (about push payments). Did this rather
than create a new issue about async communications.
* Two typos fixed

ianbjacobs committed:
Small fixes based on AHB comments

ianbjacobs committed:
More cleanup and harmonization

* General editing
* Added definition of accepted payment app
* Deleted sentence about HTTP -based mechanism since we are moving to
JS (but didn’t update the spec part itself).
* Added more ideas about optimized user experience
* Added payment_app_id alongside request_url. Thus, request_url does
not have to serve two purposes: one can identify the app one way (for
installation, filtering, etc.) and use another URL to identify an
endpoint for processing. Furthermore, we may be moving away from that
processing model so the request_url is at risk (I think).
* Moved some network failure design considerations to earlier in the
document

ianbjacobs committed:
Cleanup language about filtering per AHB

adrianhopebailie commented on index.html:
```diff
@@ -492,22 +561,23 @@
         </p>
         <p>
-            Payment apps are uniquely identifed by their <code>request_url</code>. To update the payment options
+            Payment apps are uniquely identifed by their <code>payment_app_id</code>. To update the payment options
```
Was there a decision to change this that I missed?

ianbjacobs committed:
Add pointer to issue 20; add IJ as editor

ianbjacobs committed:
Add pointer to issue 1

adrianhopebailie commented:
LGTM
